crates/hir-def/src/lib.rs

```diff
@@ -8,6 +8,7 @@
 //! actually true.
 
 #![warn(rust_2018_idioms, unused_lifetimes, semicolon_in_expressions_from_macros)]
+#![cfg_attr(feature = "in-rust-tree", feature(rustc_private))]
```

Comment:
Why is this needed?

Reply:
This feature is needed even if you import the rustc private crates indirectly, which I think is unnecessarily too aggressive. Is there any way to depend on the crates in the rustc workspace without this feature? Clippy and Miri use this, but they are fine with nightly and feature flags.

Reply:
Ah right, since we might mention the items from those crates now. Well that's unfortunate but not much we can do in that case. Would be nice  to add a comment regarding that then.